packages/x-data-grid/src/hooks/features/rowSelection/useGridRowSelection.ts

```diff
@@ -195,7 +185,11 @@ export const useGridRowSelection = (
   );
 
   const isRowSelected = React.useCallback<GridRowSelectionApi['isRowSelected']>(
-    (id) => gridRowSelectionStateSelector(apiRef.current.state).includes(id),
+    (id) => {
+      const model = gridRowSelectionStateSelector(apiRef.current.state);
+      const selectionManager = createRowSelectionManager(model);
+      return selectionManager.has(id);
```

Comment:
It would be convenient to be able to put the selection manager directly in the state rather than to have to re-create it everytime we need it :/

Reply:
Alternatively, put it in a memoized selector.

Reply:
Good point, I added `gridRowSelectionManagerSelector` and used it where applicable